authentication/config.go

```diff
@@ -98,28 +101,33 @@ func (c Config) BuildOAuthConfig(activeDirectoryEndpoint string) (*OAuthConfig,
 
 // BearerAuthorizerCallback returns a BearerAuthorizer valid only for the Primary Tenant
 // this signs a request using the AccessToken returned from the primary Resource Manager authorizer
-func (c Config) BearerAuthorizerCallback(sender autorest.Sender, oauthConfig *OAuthConfig) *autorest.BearerAuthorizerCallback {
+func (c Config) BearerAuthorizerCallback(ctx context.Context, sender autorest.Sender, oauthConfig *OAuthConfig) *autorest.BearerAuthorizerCallback {
 	return autorest.NewBearerAuthorizerCallback(sender, func(tenantID, resource string) (*autorest.BearerAuthorizer, error) {
 		// a BearerAuthorizer is only valid for the primary tenant
 		newAuthConfig := &OAuthConfig{
 			OAuth: oauthConfig.OAuth,
 		}
 
-		storageSpt, err := c.GetAuthorizationToken(sender, newAuthConfig, resource)
+		storageSpt, err := c.GetAuthorizationToken(ctx, sender, newAuthConfig, resource)
 		if err != nil {
 			return nil, err
 		}
 
 		cast, ok := storageSpt.(*autorest.BearerAuthorizer)
 		if !ok {
-			return nil, fmt.Errorf("Error converting %+v to a BearerAuthorizer", storageSpt)
+			return nil, fmt.Errorf("converting %+v to a BearerAuthorizer", storageSpt)
 		}
 
 		return cast, nil
 	})
 }
 
-// GetAuthorizationToken returns an authorization token for the authentication method defined in the Config
-func (c Config) GetAuthorizationToken(sender autorest.Sender, oauth *OAuthConfig, endpoint string) (autorest.Authorizer, error) {
-	return c.authMethod.getAuthorizationToken(sender, oauth, endpoint)
+// GetAuthorizationToken returns an autorest.Authorizer for the authentication method defined in the Config
+func (c Config) GetAuthorizationToken(ctx context.Context, sender autorest.Sender, oauth *OAuthConfig, endpoint string) (autorest.Authorizer, error) {
+	return c.authMethod.getADALToken(ctx, sender, oauth, endpoint)
+}
+
+// GetAuthorizationTokenV2 returns an autorest.Authorizer sourced from hamilton/auth
+func (c Config) GetAuthorizationTokenV2(ctx context.Context, sender autorest.Sender, oauth *OAuthConfig, endpoint string) (autorest.Authorizer, error) {
```

Comment:
can we make this:

```suggestion
// GetMSALToken returns an autorest.Authorizer using a MSAL Token via the authentication method defined in the config
func (c Config) GetMSALToken(ctx context.Context, sender autorest.Sender, oauth *OAuthConfig, endpoint string) (autorest.Authorizer, error) {
```

(this is leaking the implementation details, which will change over time)